react/src/pages/LoginCognitoBox.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import useHeader from '../hooks/Header';
+import Cookie from '../utils/Cookie';
+import Env from '../utils/Env';
+import useFetch from '../hooks/Fetch';
+import Image from '../utils/Image';
+import Server from '../utils/Server';
+import Styles from '../Styles';
+import { StandardSpinner } from '../Spinners';
+import { Auth as AmplifyAuth } from '@aws-amplify/auth'
+
+const BOX_BACKGROUND = "#FEFEFE";
+
+export const LoginCognitoBox = ({ hide }) => {
+
+    const { environ } = useParams();
+    const header = useHeader();
+
+    const config = useFetch(Server.Url("/cognito_config", false),
+    {
+        cache: true,
+        onData: (data) => {
+            const configuration = {
+                region: data.region,
+                userPoolId: data.user_pool_id,
+                userPoolWebClientId: data.client_id,
+                mandatorySignIn: true,
+                oauth: {
+                    domain: data.domain,
+                    scope: data.scope,
+                    prompt: "select_account", // DOES NOT WORK!
+                    redirectSignIn: data.callback,
+                    responseType: "code"
+                },
+            };
+            AmplifyAuth.configure(configuration);
+        }
+    });
+
+    function signinWithGoogle() {
+        Cookie.Set("env", Env.PreferredName(environ, header));
+        Cookie.Set("signinvia", "Google");
+        AmplifyAuth.federatedSignIn({ provider: "Google", prompt: 'select_account' }, { prompt: 'select_account' });
+    }
+
+    function signinWithGitHub() {
+        window.alert("Sign in with GitHub via Cognito not supported.");
+    }
+
+    const links = <div>
+        <div className="title-font" style={{marginTop:"4pt",fontSize:"8pt"}}>
+            <span>&nbsp;<a href="https://docs.aws.amazon.com/cognito/latest/developerguide/what-is-amazon-cognito.html" target="_blank" rel="noreferrer" style={{color:"gray"}}>Powered by AWS Cognito</a></span>
+            <span className="pointer" style={{float:"right"}} onClick={hide}>Cancel&nbsp;&nbsp;</span>
+        </div>
+    </div>
+
+    return <div style={{transform:"scale(1.1)",marginTop:"10pt",marginBottom:"40pt"}}>
+        <LoginCognitoBoxWrapper links={links}>
+            { config.loading ? <>
+                <StandardSpinner condition={true || config.loading} color={Styles.GetForegroundColor()} bold={false} size="140px" label={"Loading configuration "} />
+            </>:<>
+                <div style={{paddingTop:"0pt"}} />
+                <GoogleLoginButton signin={signinWithGoogle} />
+                <div style={{paddingTop:"6pt"}} />
+                <GitHubLoginButton signin={signinWithGitHub} />
```

Comment:
Are these unconditionally here or is there a way to site-configure yourself in our out of, for example, github?

Reply:
Unconditional for now. Though if you look at the GitHub one it pops up an alert saying not-yet-supported; wanted to leave it shown just as a reminder that it (if we were to go with Cognito) needs to be supported at some point.